### ISSUE TITLE
cmd/8g: doasm error compiling func == nil (386 only)

### ISSUE THREAD

**Issue by abourget** [golang/go#10194]
This simple program:

```
package main

import "fmt"

func main() {
	if Boo == nil {
		fmt.Println("Moo")
	}
}

func Boo() {
}
```

fails to compile under windows/386 with the following error:
```
gobug$ GOOS=windows GOARCH=386 go run bug.go
# command-line-arguments
./bug.go:1: doasm: notfound t2=0 from=4d to=b 00030 (/home/abourget/intel/gobug/bug.go:6)	CMPL	$"".Boo·f+0(SB),BX
```

but works well under other architectures.  I'm not sure it is valid but I needed to refactor code that used such a pattern when I tried to compile for 32 bits windows.

----
- Running on Ubuntu 14.04.2, amd64, with go version 1.4.1, downloaded from golang.org/dl

**Comment by josharian** [comment]
Reproduces with Go 1.4.2 and tip on darwin/386.

**Comment by minux** [comment]
I think this is a cmd/gc bug, that comparison should never be sent
to the backend.

**Comment by josharian** [comment]
If I hack in a workaround for #10580, I can no longer reproduce this. Waiting for a proper fix for #10580 before closing, though.

**Comment by josharian** [comment]
Figured out how to work around #10580, and I still can't reproduce this using tip. Calling this fixed until we hear otherwise.
